### PR TITLE
[ISSUE 371] Refactor Neural X Y Init

### LINKED ISSUE #371: Modify Neural Network model to use GraphML for loading neuron (x, y) locations
GraphML files will be used in the NG911 model to generate the initial graph structure (see #369). The same can be done for the Neural Network model, and we will have a unified interface for both models.

With this, we will also merge several config files (for vertex types) into one.

Tied to Workbench issue: https://github.com/UWB-Biocomputing/WorkBench/issues/131

### PR DESCRIPTION
**Pull request by Oelgazz**
closes #371

### CODE REVIEW

**Comment by jardiamj** [comment]
This looks good. There are just some empty lines and 4-space indentations that clang-format complains about in the C++ code and one small suggestion for the Python script.

**Comment by jardiamj** [comment]
I have one question about the purpose of the Python script. Is it meant to be a tool for generating the neural network graph,  with x and y locations, for a given grid size (e.g. 10x10, 100x100)? if so, it might be more useful for the script to accept the grid size,  through the command line for example.

**Comment by Oelgazz** [comment]
The purpose of the python script is to be a tool for users to generate GraphML files for the neural simulator. This script is the initial version and is updated in other branches to include parameters such as inhibitory and endogenously active neuron lists (see issues #641  and #642 ).

Because of this, it will be difficult for a user to input these details through the command line, as they will need to at least include the grid size, the inhibitory neuron list, the endogenously active neuron list, and the graph name.  Also, since using GraphML will allow the possibility of having non-grid layouts, the grid size may become a (x,y) tuple list.

**Comment by jardiamj** [comment]
> The purpose of the python script is to be a tool for users to generate GraphML files for the neural simulator. This script is the initial version and is updated in other branches to include parameters such as inhibitory and endogenously active neuron lists (see issues #641 and #642 ).
> 
> Because of this, it will be difficult for a user to input these details through the command line, as they will need to at least include the grid size, the inhibitory neuron list, the endogenously active neuron list, and the graph name. Also, since using GraphML will allow the possibility of having non-grid layouts, the grid size may become a (x,y) tuple list.

Great! That makes sense.

**Review by jardiamj — CHANGES_REQUESTED** [review]

File: Simulator/Layouts/Layout.cpp [hunk 1]
```diff
@@ -56,6 +57,10 @@ void Layout::registerGraphProperties()
 {
    // TODO: This will be implemented when all models use graphML files to load the
    // initial graph
```
[the remark this answers is not in the document]
Comment (Oelgazz) [reply Simulator/Layouts/Layout.cpp:59]: This TODO is resolved in issue #642

File: Simulator/Layouts/Neuro/LayoutNeuro.cpp [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+LayoutNeuro::LayoutNeuro() : gridLayout_(true), Layout()
 {
 }
 
-void FixedLayout::setup()
+// Register vertex properties with the GraphManager
+void LayoutNeuro::registerGraphProperties()
+{
+    // The base class registers properties that are common to all vertices
+    // TODO: Currently not fully implemented because Neuro model is still integrating graphML
```
[the remark this answers is not in the document]
Comment (Oelgazz) [reply Simulator/Layouts/Neuro/LayoutNeuro.cpp]: This TODO is resolved in issue #642

**Review by jardiamj — APPROVED** [review]

**Pull request merged**